SoObjects/Appointments/SOGoAppointmentFolder.m

```diff
@@ -1359,6 +1359,16 @@ - (void) flattenCycleRecord: (NSDictionary *) theRecord
               exRules = [component exceptionRulesWithTimeZone: tz];
             }
           
+          // Deduplicate rules
```

Comment:
You could use `uniqueObjects` from [NSArray+Utilities](SoObjects/Appointments/SOGoAppointmentFolder.m) to simplify the code.

Reply:
Will take a look when I get to it, thanks!